nsqd/stats.go

```diff
@@ -121,7 +116,9 @@ type ChannelsByName struct {
 func (c ChannelsByName) Less(i, j int) bool { return c.Channels[i].name < c.Channels[j].name }
 
 func (n *NSQD) GetStats(topic string, channel string) []TopicStats {
+	topicAcquireStart := time.Now()
 	n.RLock()
+	n.logf(LOG_DEBUG, "stats: acquiring topic list - took %v to acquire nsqd lock", time.Since(topicAcquireStart))
```

Comment:
if logging takes time is it possible to defer the log debug message? either via a defer or by moving out of the lock scope?

Reply:
Could do - I think the logger is sufficiently fast, and if the log level is higher than DEBUG its pretty much a no-op. But if we want to do this we could store the time, and defer the log write? This line is really only there temporarily to confirm this lock doesn't suffer from contention.

Reply:
It's processed at runtime, as the log level is based on configuration. So it's not strictly a noop, but it should be very fast.